pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -95,20 +111,42 @@ func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr author
 		if isAuthenticated && (isUser || isServiceAccountFromRootCluster) {
 			return a.delegate.Authorize(ctx, attr)
 		}
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, "only authenticated users and serviceaccounts in root may access root",
```

Comment:
same as above, this is a possible resolution, the reason is rather something like `toplevel.authorization.kcp.io/reason: root workspace access by non-root service account not permitted`.